doc/doxygen/CMakeLists.txt

```diff
@@ -300,6 +300,10 @@ ADD_CUSTOM_TARGET(doxygen ALL
   )
 ADD_DEPENDENCIES(documentation doxygen)
 
+# Set the canonical link for the doxygen webpages
+ADD_CUSTOM_COMMAND(TARGET doxygen POST_BUILD
+    COMMAND ${CMAKE_CURRENT_SOURCE_DIR}/scripts/set_canonical_doxygen.py)
+
```

Comment:
If the script should be run for all the documentation, how could I trigger the script with CMake? I guess that I should add the `ADD_CUSTOM_COMMAND` to `doc/CMakeList.txt`. But I don't know which should be the target/post_build?

Reply:
You should add it to the top level "documentation" target.

Reply:
I added the following command to `doc/CMakeList.txt`
```                                                                                                                                                                
ADD_CUSTOM_COMMAND(TARGET documentation POST_BUILD                                                                                                                                                               
    COMMAND ${CMAKE_CURRENT_SOURCE_DIR}/doxygen/scripts/set_canonical_doxygen.py)
```
I obtain the following error:
```
TARGET 'documentation' was not created in this directory
```
I think that `documentation` is not a target, it is a component. I don't know how CMake deals with components, but I think that `doc/CMakeList.txt` just copies the static html files. There is no "build" process. Because of that I can not add a "post_build" command.

Do you know if there is a way to filter or modify the static html files before they are copied to `${DEAL_II_DOCHTML_RELDIR}`.

Below you can find the code of `doc/CMakeList.txt`
```
INSTALL(DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}/                                                                                                                                                                 
  DESTINATION ${DEAL_II_DOCHTML_RELDIR}                                                                                                                                                                        
  COMPONENT documentation                                                                                                                                                                                      
  PATTERN "CMakeLists.txt" EXCLUDE                                                                                                                                                                             
  PATTERN "doxygen" EXCLUDE                                                                                                                                                                                    
  PATTERN "news" EXCLUDE                                                                                                                                                                                       
  )
```
I think that it is not a big deal if we can not add the canonical link to the static html files.

Another option is to add the canonical link to all the static html files of the git/GitHub repository. I do not like this option because that it is more difficult to sync. For example, if in the future we want to modify the canonical link we have to modify again all the html files by hand which is prone to errors.